server/sonar-server/src/main/java/org/sonar/server/computation/component/PathAwareCrawler.java

```diff
@@ -123,4 +139,41 @@ private T createForComponent(Component component) {
     }
   }
 
+  /**
+   * A simple object wrapping the currentPath allowing to compute the string representing the path only if
+   * the VisitException is actually built (ie. method {@link ComponentPathPrinter#toString()} is called
+   * by the internal {@link String#format(String, Object...)} of
+   * {@link VisitException#rethrowOrWrap(RuntimeException, String, Object...)}.
+   */
+  @Immutable
+  private static final class ComponentPathPrinter<T> {
+
+    private static final Joiner PATH_ELEMENTS_JOINER = Joiner.on("->");
+
+    private final DequeBasedPath<T> currentPath;
+
+    private ComponentPathPrinter(DequeBasedPath<T> currentPath) {
+      this.currentPath = currentPath;
+    }
+
+    @Override
+    public String toString() {
+      return currentPath.isRoot() ? "" : " located " + toKeyPath(currentPath);
```

Comment:
![MAJOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png) Add parentheses to make the operator precedence explicit. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](https://dory.sonarsource.com/coding_rules#rule_key=squid%3AS864)